PersistentStorage/Sources/Data/Repository/PersistentStorage/MainPersistentStorage.swift

```diff
@@ -14,154 +13,144 @@ public class MainPersistentStorage: PersistentStorage {
     ) -> AnyPublisher<T, PersistentStorageError> {
-        userDefaults
-            .publisher(for: keyPath)
-            .handleEvents(
-                receiveOutput: { [weak self] value in
-                    self?.log(
-                        message: "✅ Observing value {\(String(describing: value))}.",
-                        for: .debug
+        userDefaults(for: type)
+            .flatMap { userDefaults in
+                userDefaults
+                    .publisher(for: keyPath)
+                    .handleEvents(
+                        receiveOutput: { [weak self] value in
+                            self?.log(
+                                message: "✅ Observing value {\(String(describing: value))} in userDefaults {\(type)}",
+                                for: .debug
+                            )
+                        }
                     )
-                }
-            )
-            .flatMap { observedValue -> AnyPublisher<T, PersistentStorageError> in
-                guard let observedValue else {
-                    return Fail(error: .noValueFound).eraseToAnyPublisher()
-                }
-                return Just(observedValue)
-                    .setFailureType(to: PersistentStorageError.self)
+                    .flatMap { observedValue -> AnyPublisher<T, PersistentStorageError> in
+                        guard let observedValue else {
+                            return Fail(error: .noValueFound).eraseToAnyPublisher()
+                        }
+                        return Just(observedValue)
+                            .setFailureType(to: PersistentStorageError.self)
+                            .eraseToAnyPublisher()
+                    }
                     .eraseToAnyPublisher()
             }
             .eraseToAnyPublisher()
     }
 
     // MARK: - Internal methods
 
-    /// TODO
-    func getPersistedValue<T>(
-        valueType: T.Type,
-        valueKey: String
-    ) throws -> T {
-        guard let persistedValue = userDefaults.value(forKey: valueKey) as? T else {
-            if userDefaults.value(forKey: valueKey) != nil {
-                log(
-                    message: "❌ Value with given key exists, but method is unable to parse the value with given type.",
-                    for: .failure
-                )
-                throw PersistentStorageError.noValueFoundWithGivenType
-            } else {
-                log(
-                    message: "ℹ️ Value with given key does not exist, returning `noValueFound` failure.",
-                    for: .failure
-                )
-                throw PersistentStorageError.noValueFound
-            }
+    func userDefaults(for type: UserDefaultsType) -> AnyPublisher<UserDefaults, PersistentStorageError> {
+        guard let userDefaults = UserDefaults(type: type) else {
+            return Fail(error: PersistentStorageError.cannotCreateUserDefaults).eraseToAnyPublisher()
```

Comment:
No need to write enum name explicitly. Only `.cannotCreateUserDefaults` should be enough.